### PR TITLE
[#9096] Logic.java: Clean up unused methods

### LINKED ISSUE #9096: Logic.java: Clean up unused methods.
There are several unused methods (or methods only used in test) in `Logic.java`. See if we can remove them as well as their correspoding methods `*Db`.

### PR DESCRIPTION
**Pull request by IAmPramod**
Fixes #9096

**Outline of Solution**

Removed unused methods from Logic.java.
Modified test cases accordingly.

### CODE REVIEW

**Comment by IAmPramod** [comment]
@xpdavid made the change. Ready for review!

File: src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java [hunk 1]
```diff
@@ -264,10 +264,10 @@ public void testUpdateFeedbackResponseComment() throws Exception {
...
 
     @Test
-    public void testDeleteFeedbackResponseComment() throws Exception {
+    public void testDeleteFeedbackResponseCommentById() throws Exception {
         //create a frComment to delete
         FeedbackResponseCommentAttributes frComment = restoreFrCommentFromDataBundle("comment1FromT1C1ToR1Q1S1C1");
-        frComment.setId(null);
+        frComment.setId(1L);
         frComment.feedbackQuestionId = getQuestionIdInDataBundle("qn2InSession1InCourse1");
         frComment.feedbackResponseId = getResponseIdInDataBundle("response2ForQ2S1C1", "qn2InSession1InCourse1");
```
Comment (xpdavid) [inline src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: I don't think you need L270 - L272

Comment (IAmPramod) [reply src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: @xpdavid We need to retain L270, else there will be NPE at L281. Other two lines can be removed.

Comment (xpdavid) [reply src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: But why you need `frComment.getId()`? You are testing `deleteFeedbackResponseCommentById` under the scenario that `silent fail nothing to delete`. So ideally, you can just supply an non-exist ID let's say `1234567`.

A more sophisticated method would be to use `getFeedbackResponseComment` to make sure `1234567` doesn't exist before the test.

Comment (IAmPramod) [reply src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: @xpdavid got it now.
Will modify this accordingly.

**Comment by xpdavid** [comment]
Oh.. and I believe there are other unused methods, pls find them by yourself.

e.g. `public void createStudent(StudentAttributes student)` `public void createStudentWithoutDocument(StudentAttributes student)`

**Comment by xpdavid** [comment]
If you think the changes are too big, we can do it in several PR. :)

Maybe we should focus on the four methods first?

**Comment by IAmPramod** [comment]
I think we should focus on the four methods first.
I will raise separate PR's for other methods .

File: src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java (hunk 1 above)
Comment (IAmPramod) [reply src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: @xpdavid I have made the change.

File: src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java [hunk 2]
```diff
@@ -93,7 +93,7 @@ public void testCreateFeedbackResponseComment() throws Exception {
         assertNotNull(actualFrComment);
 
         //delete afterwards
-        frcLogic.deleteFeedbackResponseComment(frComment);
+        frcLogic.deleteFeedbackResponseCommentById(frComment.getId());
```
Comment (xpdavid) [inline src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: Are you sure that `frComment.getId()` return an real ID?

File: src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java [hunk 3]
```diff
@@ -278,7 +275,7 @@ public void testDeleteFeedbackResponseComment() throws Exception {
         frComment.feedbackResponseId = "invalid responseId";
         //without proper frCommentId and its feedbackResponseId,
```
Comment (xpdavid) [inline src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: Please delete invalid comment and statement :)

File: src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java [hunk 4]
```diff
@@ -264,12 +264,9 @@ public void testUpdateFeedbackResponseComment() throws Exception {
...
     @Test
-    public void testDeleteFeedbackResponseComment() throws Exception {
+    public void testDeleteFeedbackResponseCommentById() throws Exception {
         //create a frComment to delete
         FeedbackResponseCommentAttributes frComment = restoreFrCommentFromDataBundle("comment1FromT1C1ToR1Q1S1C1");
-        frComment.setId(null);
-        frComment.feedbackQuestionId = getQuestionIdInDataBundle("qn2InSession1InCourse1");
-        frComment.feedbackResponseId = getResponseIdInDataBundle("response2ForQ2S1C1", "qn2InSession1InCourse1");
 
         frcLogic.createFeedbackResponseComment(frComment);
```
Comment (xpdavid) [inline src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: Why you need to create the comment?

File: src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java [hunk 5]
```diff
@@ -278,7 +275,7 @@ public void testDeleteFeedbackResponseComment() throws Exception {
         frComment.feedbackResponseId = "invalid responseId";
         //without proper frCommentId and its feedbackResponseId,
         //it cannot be deleted
-        frcLogic.deleteFeedbackResponseComment(frComment);
+        frcLogic.deleteFeedbackResponseCommentById(1234567L);
 
         FeedbackResponseCommentAttributes actualFrComment =
```
Comment (xpdavid) [inline src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: The purpose of the following verification is?

File: src/test/java/teammates/test/cases/logic/FeedbackResponsesLogicTest.java [hunk 6]
```diff
@@ -155,22 +156,6 @@ private void testUpdateFeedbackResponse() throws Exception {
             AssertHelper.assertContains("Trying to create a Feedback Response that exists", e.getMessage());
         }
 
-        ______TS("success: standard update with carried params - using createFeedbackResponse");
```
[the remark this answers is not in the document]
Comment (xpdavid) [reply src/test/java/teammates/test/cases/logic/FeedbackResponsesLogicTest.java]: Check that if we have same test for `createFeedbackResponse(List xx)`, if not, you need to turn this test into to use `createFeedbackResponse(List xx)` instead.

File: src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java (hunk 4 above)
Comment (IAmPramod) [reply src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: Yes, we need not create this comment. I will remove this now.

File: src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java (hunk 3 above)
Comment (IAmPramod) [reply src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: yes, removing this.

File: src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java (hunk 2 above)
Comment (IAmPramod) [reply src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: Yes, this is a real Id. I verified this using `verifyPresentInDatastore` and `verifyAbsentInDatastore` before and after this execution. I will add `verifyAbsentInDatastore` after this to be sure.

Comment (xpdavid) [reply src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: Can just use `frcLogic.getFeedbackResponseComment()` to verify.

File: src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java (hunk 5 above)
Comment (IAmPramod) [reply src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: Yes, This is not needed here. I think we can safely remove test case `silent fail nothing to delete`. Your thoughts in this?

Comment (xpdavid) [reply src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: No we still need that test case but you don't need to add any verification. You can think of the verification as 'It doesn't throw exception'.

File: src/test/java/teammates/test/cases/logic/FeedbackResponsesLogicTest.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (IAmPramod) [reply src/test/java/teammates/test/cases/logic/FeedbackResponsesLogicTest.java]: @xpdavid  We don't have same test for `createFeedbackResponses`. I tried adding the same test and it doesn't seem to update to existing feedback responses.

I looked at the logic and here are my findings.

- `createFeedbackResponses(List<FeedbackResponseAttributes>)` uses `createEntities` to create entities;
- `createEntities` in `EntitiesDb` at L63 creates the new entities and returns a list of entities which already exists. We are not doing anything with this list in `createFeedbackResponses(List<FeedbackResponseAttributes>)`.
- There's a method in `FeedbackResponsesDb` at L35 `createFeedbackResponses` which does the save and update of the entities, but I don't see it getting used anywhere in `FeedbackResponsesLogic`.

Am i missing something here?

[the remark this answers is not in the document]
Comment (xpdavid) [reply src/test/java/teammates/test/cases/logic/FeedbackResponsesLogicTest.java]: Sure. Then no need to add that.

[the remark this answers is not in the document]
Comment (IAmPramod) [reply src/test/java/teammates/test/cases/logic/FeedbackResponsesLogicTest.java]: @xpdavid I'm curious about the functionality intended for `createFeedbackResponses(List<FeedbackResponseAttributes>)`. Is it supposed to edit existing feedbackResponses?

[the remark this answers is not in the document]
Comment (xpdavid) [reply src/test/java/teammates/test/cases/logic/FeedbackResponsesLogicTest.java]: I don't think so. It is only used in `FeedbackSubmissionEditSaveAction#saveNewResponses`

[the remark this answers is not in the document]
Comment (IAmPramod) [reply src/test/java/teammates/test/cases/logic/FeedbackResponsesLogicTest.java]: ok thanks.

**Comment by IAmPramod** [comment]
@xpdavid Made the changes, Ready for review.

File: src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java (hunk 2 above)
Comment (xpdavid) [reply src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: Wait, In L73, the ID is set to `null`.

File: src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java (hunk 4 above)
Comment (xpdavid) [reply src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: You forget to remove?

File: src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java [hunk 7]
```diff
@@ -264,30 +265,22 @@ public void testUpdateFeedbackResponseComment() throws Exception {
...
 
         frcLogic.createFeedbackResponseComment(frComment);
 
         ______TS("silent fail nothing to delete");
 
-        frComment.feedbackResponseId = "invalid responseId";
-        //without proper frCommentId and its feedbackResponseId,
-        //it cannot be deleted
-        frcLogic.deleteFeedbackResponseComment(frComment);
+        frcLogic.deleteFeedbackResponseCommentById(1234567L);
```
Comment (xpdavid) [inline src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: You need to verify `1234567L` doesn't exist first.

File: src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java [hunk 8]
```diff
@@ -264,30 +265,22 @@ public void testUpdateFeedbackResponseComment() throws Exception {
...
                                  frComment.courseId, frComment.feedbackSessionName).get(1);
-        verifyPresentInDatastore(actualFrComment);
-
-        ______TS("typical success case");
-
-        frcLogic.deleteFeedbackResponseComment(actualFrComment);
+        frcLogic.deleteFeedbackResponseCommentById(actualFrComment.getId());
         verifyAbsentInDatastore(actualFrComment);
 
         ______TS("typical success case for response");
```
Comment (xpdavid) [inline src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: This test doesn't test `deleteFeedbackResponseCommentById`

Comment (IAmPramod) [reply src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: yes, moved this to a new test for `deleteFeedbackResponseCommentsForResponse`

File: src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java (hunk 7 above)
Comment (IAmPramod) [reply src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: Added `assertNull(frcLogic.getFeedbackResponseComment(1234567L))` before this.

**Comment by IAmPramod** [comment]
@xpdavid Ready for review,

File: src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java (hunk 2 above)
Comment (xpdavid) [reply src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: Actually there is a problem with the test. I'll create a separate issue for it. Currently can use `frComment.getId()`

File: src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java [hunk 9]
```diff
@@ -264,40 +265,33 @@ public void testUpdateFeedbackResponseComment() throws Exception {
     }
 
     @Test
-    public void testDeleteFeedbackResponseComment() throws Exception {
+    public void testDeleteFeedbackResponseCommentById() throws Exception {
         //create a frComment to delete
         FeedbackResponseCommentAttributes frComment = restoreFrCommentFromDataBundle("comment1FromT1C1ToR1Q1S1C1");
```
Comment (xpdavid) [inline src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: `//create a frComment to delete` The comment is not valid any more.

In addition, move the variable declaration to the place where it is first used.

File: src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java (hunk 2 above)
Comment (IAmPramod) [reply src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: ok

File: src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java (hunk 9 above)
Comment (IAmPramod) [reply src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: @xpdavid which variable you are talking about?

Comment (xpdavid) [reply src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: `frComment `

Comment (IAmPramod) [reply src/test/java/teammates/test/cases/logic/FeedbackResponseCommentsLogicTest.java]: Made the change.

**Comment by IAmPramod** [comment]
@xpdavid Ready for review.

**Comment by IAmPramod** [comment]
@xpdavid gentle reminder!

File: src/test/java/teammates/test/cases/logic/FeedbackResponsesLogicTest.java [hunk 10]
```diff
@@ -362,7 +347,7 @@ private int getResponseRate(String sessionName, String courseId) {
 
     private void restoreStudentFeedbackResponseToDatastore(FeedbackResponseAttributes response)
             throws InvalidParametersException, EntityDoesNotExistException {
-        frLogic.createFeedbackResponse(response);
+        frLogic.createFeedbackResponses(Arrays.asList(response));
```
[the remark this answers is not in the document]
Comment (IAmPramod) [reply src/test/java/teammates/test/cases/logic/FeedbackResponsesLogicTest.java]: Yeah,

Shall i add a check if feedback response already exists, update that else create new?

[the remark this answers is not in the document]
Comment (xpdavid) [reply src/test/java/teammates/test/cases/logic/FeedbackResponsesLogicTest.java]: You can add a method in the test case to do the job.

**Comment by xpdavid** [comment]
@IAmPramod any update?

**Comment by IAmPramod** [comment]
@xpdavid ready for review.

**Pull request merged**